src/handlers/admin/menu_handler.py

```diff
@@ -10,187 +11,103 @@
+    query = update.callback_query
+    data = json.loads(query.data)
+    
+    action_handlers = {
+        constants.Actions.start_select_chat: handle_start_select_chat,
+        constants.Actions.select_chat: handle_select_chat,
+        constants.Actions.get_current_settings: handle_get_current_settings,
+    }
+    
+    if data["action"] in action_handlers:
+        action_handlers[data["action"]](data, query, context)
+    
     elif data["action"] in [
         constants.Actions.set_on_new_chat_member_message_response,
         constants.Actions.set_kick_timeout,
-        constants.Actions.set_notify_message,
-        constants.Actions.set_on_known_new_chat_member_message_response,
-        constants.Actions.set_on_successful_introducion_response,
-        constants.Actions.set_on_kick_message,
+        # ... Other actions ...
     ]:
-        context.bot.edit_message_text(
-            text="Отправьте новое значение",
-            chat_id=query.message.chat_id,
-            message_id=query.message.message_id,
-        )
-        context.user_data["chat_id"] = data["chat_id"]
-        context.user_data["action"] = data["action"]
-
-    elif data["action"] == constants.Actions.get_current_settings:
-        keyboard = [
-            [
-                InlineKeyboardButton(
-                    "К настройке чата",
-                    callback_data=json.dumps(
-                        {
-                            "chat_id": data["chat_id"],
-                            "action": constants.Actions.select_chat,
-                        }
-                    ),
-                ),
-                InlineKeyboardButton(
-                    "К списку чатов",
-                    callback_data=json.dumps(
-                        {"action": constants.Actions.start_select_chat}
-                    ),
-                ),
-            ],
-        ]
-
-        reply_markup = InlineKeyboardMarkup(keyboard)
-        with session_scope() as sess:
-            chat = sess.query(Chat).filter(Chat.id == data["chat_id"]).first()
-            context.bot.edit_message_text(
-                text=constants.get_settings_message.format(**chat.__dict__),
-                parse_mode=ParseMode.MARKDOWN,
-                chat_id=query.message.chat_id,
-                message_id=query.message.message_id,
-                reply_markup=reply_markup,
-            )
-
-        context.user_data["action"] = None
-
+        handle_action(data, query, context)
+
+def set_chat_attribute(chat_id, attribute, value):
+    with session_scope() as sess:
+        chat = Chat(id=chat_id, **{attribute: value})
+        sess.merge(chat)
+
+def send_message_with_keyboard(update, message_text, chat_id=None):
+    keyboard = [
+        [
+            new_button("К настройке чата", constants.Actions.select_chat, chat_id),
+            new_button("К списку чатов", constants.Actions.start_select_chat),
+        ],
+    ]
+    reply_markup = InlineKeyboardMarkup(keyboard)
```

Comment:
we can put the keyboard creation from a (nested -- to allow both vertical and horizontal layout) list of buttons into a function (e.g. it could get the argument of type `Union[List[List[Button]], List[Button]]`) and return an `InlineKeyboardMarkup` object